projects/cm_mcu/LocalTasks.c

```diff
@@ -335,20 +339,87 @@ struct MonitorI2CTaskArgs_t clockr0a_args = {
     .stack_size = 4096U,
 };
 
+void setFFmask(uint32_t ff_combined_mask)
```

Comment:
as far as I can tell what this code does is invert the bits in the `ff_combined_mask`. is that the intention?

Reply:
I think the present signal must be inverted in order to get the correct mask. So, that is intentional.

Reply:
ok, if you want to invert a 20 bit int you can just do it with this piece of code:
```
uint32_t newval = (~oldval) & 0xFFFFFU; 
```
Having said that, why do you want to invert it? the PRESENT signals are 1 for present, 0 for absent. Is that not what we want?

Reply:
Okay, I will invert it that way. 
re: why invert ? the PRESENT signals are actually 1 for absent and 0 for present.

Reply:
ah ok I looked at the schematic again and you are right, they are written as `/XYZ_PRESENT`. The leading slash is the convention for an active low signal. I had missed that.

Reply:
sorry, I think the part of inverting no longer exists and is revised to https://github.com/apollo-lhc/cm_mcu/blob/first_time_mcu/projects/cm_mcu/LocalTasks.c#L352

Reply:
that's the code I was referring to...

Reply:
Okay. Done.

Reply:
I hate to be a PITA but now you are doing a loop which copies `ff_combined_mask` to `data` without making any changes, and then inverts `data`. Why not just skip the loop and use `ff_combined_mask` directly? Or add a comment explaining why you're doing it this way so that future you and me (and others) are not confused...

Reply:
That's true. I was not aware of this. It will just be `uint32_t newval = (~oldval) & 0xFFFFFU;` without looping. I will fix it and add a comment shortly.